src/main/java/io/cronitor/client/CronitorPinger.java

```diff
@@ -0,0 +1,81 @@
+package io.cronitor.client;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.logging.Logger;
+
+import io.cronitor.client.urlgenerators.CommandUrlGenerator;
+
+public class CronitorPinger {
+
+    public CronitorPinger() {
+
+        this.connection = null;
+    }
+
+
+
+    HttpURLConnection connection;
+    URL url;
+    private final static Logger logger = Logger.getLogger(CronitorPinger.class.getName());
+    private final Integer cronitorPingTimeoutInSecond = 5;
+
+    public void ping(String command, String monitorCode, String apiKey, String message) throws IOException, URISyntaxException {
+        for (int i=0; i<4; i++) {
+            Boolean usePrimaryPingDomain = i < 2;
```

Comment:
This line can be removed and the following one replaced by setConnection(getURL((i < 2), command, monitorCode, apiKey, message));